src/store/actions.js

```diff
@@ -75,123 +128,65 @@ export default {
 
   // items
 
-  _addItem({ commit, dispatch, getters }, newItem) {
-    commit('addItem', {
-      currentListIndex: getters.currentListIndex,
-      newItem,
-    });
-    dispatch('_setChangingStatus', false);
-    dispatch('_updateLocal');
-  },
-  _setItems({ commit, dispatch, getters }, newItems) {
-    commit('setItems', {
-      currentListIndex: getters.currentListIndex,
-      newItems,
-    });
-    dispatch('_setChangingStatus', false);
-    dispatch('_updateLocal');
-  },
-  _deleteItem({ commit, dispatch, getters }, item) {
-    if (item) {
-      commit('deleteItem', {
-        currentListIndex: getters.currentListIndex,
-        id: item.id,
-      });
-      dispatch('_updateLocal');
-    }
+  _addItem({ commit, getters }, item) {
+    const listId = getters.currentListObj.id;
 
-    dispatch('_setItemForEditting', null);
+    return this._vm.$axios.post(`${this._vm.$apiBasePath}item/add/${listId}`, item)
+      .then(response => {
+        commit('addItem', response.data);
+      })
+      .catch(error => {
+        throw error;
+      });
   },
-  _setItemForEditting({ commit, dispatch }, item) {
-    commit('setItemForEditting', item);
-    dispatch('_setChangingStatus', true);
-
-    if (!item) {
-      dispatch('_setChangingStatus', false);
-    }
+  _updateItem({ commit, dispatch }, item) {
+    const itemDataToSend = {
+      text: item.text,
+      details: item.details,
+      tags: item.tags,
+      category: item.category,
+    };
```

Comment:
And why do you use this const if we use it just once?
What do you think if we just put it in argument?